File: api/src/main/java/com/makingiants/today/api/utils/TextUtils.java
package com.makingiants.today.api.utils;

import android.support.annotation.IntDef;
import android.support.annotation.Nullable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Used to check string values.
 */
public final class TextUtils {

  public static final int ERROR_EMPTY = 0;

  @IntDef(value = { ERROR_EMPTY })
  @Retention(RetentionPolicy.SOURCE)
  public @interface ValidationError {
  }

  private TextUtils() {

  }

  public static boolean isEmpty(@Nullable CharSequence charSequence) {
    return charSequence == null || charSequence.toString().trim().isEmpty();
  }
}

File: api/src/main/kotlin/com/makingiants/today/api/utils/TextUtils.kt
package com.makingiants.today.api.utils

import android.support.annotation.IntDef

/**
 * Used to check string values.
 */
object TextUtils {
  const val ERROR_EMPTY = 0L

  @IntDef(ERROR_EMPTY)
  @Retention(AnnotationRetention.SOURCE)
  annotation class ValidationError

  fun isEmpty(charSequence: CharSequence?): Boolean {
    return charSequence == null || charSequence.toString().trim { it <= ' ' }.isEmpty()
  }
}
